### PR TITLE
Conversions: Case sensitive Units/Symbol handling

### PR DESCRIPTION
**Pull request by mintsoft**
## Description of new Instant Answer, or changes

This changes the behaviour of conversions as follows:
* Descriptive unit names (microseconds, megabytes, megabits, kilometers etc) should remain case insensitive, if the user puts in MICROSECONDS because they love capslock then it should work
* Things that I've defined as Symbols (based on SI definitions, i.e. `m` for metres, `l` for Litres, `Pa` for Pascals, and crucially: `B` for bytes and `b` for bits) should be *case sensitive* as this is how the symbol is defined. 
* Symbol case sensitivity is also extended to the SI prefixes i.e. `Mm` is megametres and `mm` is millimeters, similiarily `kB` is kilobytes and `kb` is kilobits

I've also extended the volume bodge to add more unit support so that more queries using `oz` as implicitly `fl oz` work

## Related Issues and Discussions
Fixes: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/3990
Fixes: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1396

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [x] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [x] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [x] There is no unnecessary files in place (such as editor config files)
- [x] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
Deploying to Beta...

**Comment by mintsoft** [comment]
Examples that do work:
 https://duckduckgo.com/?q=1+megabyte+in+kilobytes&t=ffab&ia=answer
 https://beta.duckduckgo.com/?q=1+MB+to+kB&t=hd&ia=answer
 https://beta.duckduckgo.com/?q=1+MiB+to+kb&t=hd&ia=answer

I *can* alias mb to megabytes even though it should be millibits (0.001 bits); however millibits is a nonsense unit, I don't anticipate there being any direct issue with us not supporting it in favour of mega. 

That said, the problem comes with the [Bb]. So following what's alluded to in: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/3990 and combining with the observation from @moollaza I think the following "makes sense":

* `Mb = mb = megabits`
* `MB = mB = megabytes`

Basically I'm losing the case sensitivity on the M vs m, but only for digital information units. For all other physical quantities m will still mean 10^-3 and M will mean 10^6.

Upshot would be, that the query moollaza tried would be interpreted as `1 megabit in kilobits = 1000`

**Comment by moollaza** [comment]
@mintsoft this is live! However it seems like `mb`, `gb`, etc. no longer work? 

https://duckduckgo.com/?q=40+gb+to+mb&ia=web

I think we should be assuming Megabytes and Gigabytes here?

**Comment by mintsoft** [comment]
@moollaza I raised this the other day: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/4073 It needs the 
    Mb = mb = megabits
    MB = mB = megabytes
rules implementing. I'll be doing them this evening hopefully
